server.js

```diff
@@ -827,23 +827,27 @@ cache(function(data, match, sendBadge, request) {
 }));
 
 // SonarQube code coverage
-camp.route(/^\/sonar\/(http|https)\/(.*)\/(.*)\/(.*)\.(svg|png|gif|jpg|json)$/,
+camp.route(/^\/sonar\/?(.*)?\/(http|https)\/(.*)\/(.*)\/(.*)\.(svg|png|gif|jpg|json)$/,
     cache(function(data, match, sendBadge, request) {
-      var scheme = match[1];
-      var serverUrl = match[2];  // eg, `sonar.qatools.ru`.
-      var buildType = match[3];  // eg, `ru.yandex.qatools.allure:allure-core:master`.
-      var metricName = match[4];
-      var format = match[5];
+      var version = parseFloat(match[1]);
+      var scheme = match[2];
+      var serverUrl = match[3];
+      var buildType = match[4];
+      var metricName = match[5];
+      var format = match[6];
 
       var sonarMetricName = metricName;
       if (metricName === 'tech_debt') {
         //special condition for backwards compatibility
         sonarMetricName = 'sqale_debt_ratio';
       }
 
```

Comment:
Is it worth checking that `version` is not NaN and returning an error? They don't have point releases, but they do have suffixes. I'm not quite sure what this code will do when passed NaN.

Alternatively, I wonder if we have a version parsing function somewhere…

Reply:
do we have something like [`validators.js`](https://github.com/badges/shields/blob/master/service-tests/helpers/validators.js) for `server.js`?

Reply:
I'm fairly certain that if it was `NaN` then the latest version would be used as `!!NaN == false`

Reply:
Will i need to validate that this is a version once the regex has been changed to`[0-9.]+` as `parseFloat('1.2.345'` would return `1.2` so it should always be valid.

Reply:
> do we have something like [`validators.js`](https://github.com/badges/shields/blob/master/service-tests/helpers/validators.js) for `server.js`?

No, though I think I'd like to address validation as part of the service rewrites.

No need to stress it right now.